src/components/tabs/tabs.js

```diff
@@ -269,19 +270,46 @@ class Tabs extends React.Component {
   }
 
   /**
-   * Handles the changing of tabs
+   * Handles the changing of tabs with the mouse
    *
    * @method handleTabClick
    * @param {Event} ev Click Event
    */
   handleTabClick = (ev) => {
-    if (Event.isEnterKey(ev) || !Event.isEventType(ev, 'keydown')) {
-      const tabid = ev.target.dataset.tabid;
-      this.updateVisibleTab(tabid);
-    }
+    if (Event.isEventType(ev, 'keydown')) { return; }
+    const tabid = ev.target.dataset.tabid;
+    this.updateVisibleTab(tabid);
   }
 
+  /**
+   * Handles the keyboard navigation of tabs
+   *
+   * @method handleKeyDown
+   * @param {Number} index Index of the tab
+   */
+  handleKeyDown = (index) => {
+    return (event) => {
+      event.stopPropagation();
+      if (Event.isEnterKey(event)) {
+        this.updateVisibleTab(this.tabIds[index]);
+      } else if (Event.isLeftKey(event)) {
+        this.updateVisibleTab(this.tabIds[index - 1]);
+        this.focusTab(this[this.tabRefs[index - 1]]);
+      } else if (Event.isRightKey(event)) {
```

Comment:
does this need to be up and down when tabs are vertical?

Reply:
Just committed this in: https://github.com/Sage/carbon/pull/1528/commits/2ab3937493f210020c3cbf72a2b55946b1aaa4a7